File: Modul 6/soal_5/PRAK605-2410817310003-Muhammad_Guntur_Ricky_Adhitya.py
def matriks(ordo):
    matriks_temp = list()
    for i in range(0, ordo) :
        vektor_i = list(map(int, input().split()))
        matriks_temp.append(vektor_i)
    return matriks_temp

def transpose(ordo, matriks):
    matriks_temp = list()
    for i in range(0, ordo):
        vektor_i = list()
        for j in range(0, ordo):
            vektor_i.append(matriks[j][i])
        matriks_temp.append(vektor_i)
    return matriks_temp

def kali_vektor(vektor_a, vektor_b, ordo):
    hasil = 0
    for i in range(0, ordo):
        hasil += vektor_a[i] * vektor_b[i]
    return hasil

def print_matriks(matriks):
    for baris in matriks :
        for elemen in baris:
            print(elemen, end=" ")
        print()

ordo = int(input())

print("Matriks A")
matriks_a = matriks(ordo)

print("Matriks B")
matriks_b = matriks(ordo)

matriks_b = transpose(ordo, matriks_b)

hasil = []

print("Matriks AXB")
for i in range(0, ordo):
    vektor_i = []
    for j in range(0, ordo):
        operasi = kali_vektor(matriks_a[i], matriks_b[j], ordo)
        vektor_i.append(operasi)
    hasil.append(vektor_i)

print_matriks(hasil)

File: Modul 6/soal_5/PRAK605-2410817310003-Muhammad_Guntur_Ricky_Adhitya.c
#include <stdio.h>

int ordo_global = 0;

int vektor_kali(int vektor_a[], int vektor_b[]){
    int hasil = 0;
    for(int i = 0; i < ordo_global; ++i){
        hasil += vektor_a[i] * vektor_b[i];
    }
    return hasil;  
}

void transpose_matriks(int matriks[][ordo_global], int ordo){
    int matriks_baru[ordo][ordo];
    for(int i = 0; i < ordo; ++i){
        for(int j = 0; j < ordo; ++j){
            matriks_baru[i][j] = matriks[j][i];
        }
    }

    for(int i = 0; i < ordo; ++i){
        for(int j = 0; j < ordo; ++j){
            matriks[i][j] = matriks_baru[i][j];
        }
    }
}

void scan_matriks(int matriks[][ordo_global], int ordo){
    for(int i = 0; i < ordo; ++i){
        for(int j = 0; j < ordo; ++j){
            scanf("%d", &matriks[i][j]);
        }
    }
}

void print_matriks(int matriks[][ordo_global], int ordo){
    for(int i = 0; i < ordo; ++i){
        for(int j = 0; j < ordo; ++j){
            printf("%d ", matriks[i][j]);
        }
        printf("\n");
    }
}

void perkalian_matriks(int ordo){
    int matriks_a[ordo][ordo];
    printf("Matriks A\n");
    scan_matriks(matriks_a, ordo);
    
    int matriks_b[ordo][ordo];
    printf("Matriks B\n");
    scan_matriks(matriks_b, ordo);

    transpose_matriks(matriks_b, ordo);

    int matriks_hasil[ordo][ordo];
    for(int row = 0; row < ordo; ++row){
        for(int collumn = 0; collumn < ordo; ++collumn){
            matriks_hasil[row][collumn] = vektor_kali(matriks_a[row], matriks_b[collumn]);
        }
    }

    printf("Matriks AXB\n");
    print_matriks(matriks_hasil, ordo);
}

int main(){
    scanf("%d", &ordo_global);

    perkalian_matriks(ordo_global);

    return 0;
}